javalib/src/main/scala/java/util/HashMap.scala

```diff
@@ -180,6 +180,50 @@ class HashMap[K, V](initialCapacity: Int, loadFactor: Double)
     }
   }
 
+  override def computeIfAbsent(key: K, mappingFunction: Function[_ >: K, _ <: V]): V = {
+    val (node, hash, idx, oldValue) = getNode0(key)
+    if (oldValue != null) {
+      oldValue
+    } else {
+      val newValue = mappingFunction.apply(key)
+      if (newValue != null)
+        put0(key, newValue, hash, node)
+      newValue
+    }
+  }
+
+  override def computeIfPresent(key: K, remappingFunction: BiFunction[_ >: K, _ >: V, _ <: V]): V = {
+    val (node, _, idx, oldValue) = getNode0(key)
+    if (oldValue == null) {
+      oldValue
+    } else {
+      val newValue = remappingFunction.apply(key, oldValue)
+      if (newValue != null)
+        node.value = newValue
+      else
+        remove0(node, idx)
+      newValue
+    }
+  }
+
+  override def compute(key: K, remappingFunction: BiFunction[_ >: K, _ >: V, _ <: V]): V = {
+    val (node, hash, idx, oldValue) = getNode0(key)
+    val newValue = remappingFunction.apply(key, oldValue)
+    putOrRemove0(key, hash, idx, node, newValue)
+    newValue
```

Comment:
Consider returning ` newValue` form `putOrRemove0` (seems all the call sites do that anyways).